### PR TITLE
[6.x] Support intended URLs in EnsureEmailIsVerified

### PR DESCRIPTION
**Pull request by amayer5125**
This change will be helpful for developers that need someone redirected back to an intended url after registering a user account. Similar to how they are redirected after authenticating.

To achieve this I refactored the EnsureEmailIsVerified middleware to match code in RequirePassword middleware. The intended url is kept intact through the email verification process as well.

The refactor has the added benefit of returning json resource when json is requested instead of using abort helper. Which I believe is a bug and I reported in #30663 .

### CODE REVIEW

File: src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php [hunk 1]
```diff
@@ -4,28 +4,73 @@
...
+        $this->responseFactory = $responseFactory;
+        $this->urlGenerator = $urlGenerator;
+    }
+
     /**
      * Handle an incoming request.
      *
      * @param  \Illuminate\Http\Request  $request
      * @param  \Closure  $next
-     * @param  string|null  $redirectToRoute
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: Please revert this incorrect change.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php [hunk 2]
```diff
@@ -4,28 +4,73 @@
 
 use Closure;
 use Illuminate\Contracts\Auth\MustVerifyEmail;
-use Illuminate\Support\Facades\Redirect;
+use Illuminate\Contracts\Routing\ResponseFactory;
+use Illuminate\Routing\UrlGenerator;
```
Comment (X-Coder264) [inline src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: The import should be changed to `Illuminate\Contracts\Routing\UrlGenerator` as the docblock below correctly states.

Comment (amayer5125) [reply src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: Updated!

File: src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php (hunk 1 above)
Comment (amayer5125) [reply src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: Reverted

File: src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php [hunk 3]
```diff
@@ -4,28 +4,73 @@
...
+    }
+
     /**
      * Handle an incoming request.
      *
      * @param  \Illuminate\Http\Request  $request
      * @param  \Closure  $next
      * @param  string|null  $redirectToRoute
-     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     * @return mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: Please keep the original return values, and make sure the implementation returns responses.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php (hunk 3 above)
Comment (amayer5125) [reply src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: Are you asking me to just revert this line? I believe all returns are responses or redirects.

Comment (GrahamCampbell) [reply src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: Yes, then.

Comment (amayer5125) [reply src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php]: Line reverted.

**Comment by taylorotwell** [comment]
No plans to change this on the 6.x release.

**Pull request closed without merging**